packages/@aws-cdk/aws-codepipeline/lib/cross-region-support-stack.ts

```diff
@@ -28,20 +28,39 @@ function lastNCharacters(str: string, n: number) {
   return str.substring(startIndex);
 }
 
+/**
+ * Props for the support stack
+ */
+export interface CrossRegionSupportConstructProps {
+  /**
+   * Whether to create the KMS CMK
+   *
+   * (Required for cross-account deployments)
+   *
+   * @default true
+   */
+  readonly createKmsKey?: boolean;
+}
+
 export class CrossRegionSupportConstruct extends cdk.Construct {
   public readonly replicationBucket: s3.IBucket;
 
-  constructor(scope: cdk.Construct, id: string) {
+  constructor(scope: cdk.Construct, id: string, props: CrossRegionSupportConstructProps = {}) {
     super(scope, id);
 
-    const encryptionKey = new kms.Key(this, 'CrossRegionCodePipelineReplicationBucketEncryptionKey', {
-      removalPolicy: cdk.RemovalPolicy.DESTROY,
-    });
-    const encryptionAlias = new AliasWithShorterGeneratedName(this, 'CrossRegionCodePipelineReplicationBucketEncryptionAlias', {
-      targetKey: encryptionKey,
-      aliasName: cdk.PhysicalName.GENERATE_IF_NEEDED,
-      removalPolicy: cdk.RemovalPolicy.DESTROY,
-    });
+    const createKmsKey = props.createKmsKey ?? true;
+
+    let encryptionAlias;
+    if (createKmsKey) {
+      const encryptionKey = new kms.Key(this, 'CrossRegionCodePipelineReplicationBucketEncryptionKey', {
+        removalPolicy: cdk.RemovalPolicy.DESTROY,
+      });
+      encryptionAlias = new AliasWithShorterGeneratedName(this, 'CrossRegionCodePipelineReplicationBucketEncryptionAlias', {
+        targetKey: encryptionKey,
+        aliasName: cdk.PhysicalName.GENERATE_IF_NEEDED,
+        removalPolicy: cdk.RemovalPolicy.DESTROY,
+      });
+    }
     this.replicationBucket = new s3.Bucket(this, 'CrossRegionCodePipelineReplicationBucket', {
       bucketName: cdk.PhysicalName.GENERATE_IF_NEEDED,
       encryptionKey: encryptionAlias,
```

Comment:
If `encryptionAlias` is `undefined`, should the `encryption` property of this Bucket be `KMS_MANAGED`? That's how it's done in the CodePipeline Bucket.